komorebi-bar/src/main.rs

```diff
@@ -234,9 +234,24 @@ fn main() -> color_eyre::Result<()> {
         std::process::exit(0);
     }
 
-    let state = serde_json::from_str::<komorebi_client::State>(&komorebi_client::send_query(
-        &SocketMessage::State,
-    )?)?;
+    let retry_duration = Duration::from_secs(1);
```

Comment:
I don't think we can retry in a loop here -  we need to bail out if something isn't working to avoid spinning cycles and provide some form of user feedback that they can share when they open a support thread

Reply:
I agree, I don't think this is ideal. I had to make this change because the bars would now launch too fast, crashing as `komorebi` is yet ready to accept connections. If there is a way to know when `komorebi` has completed its startup and then we launch the bars that would be more ideal, but I don't know of a nice way of implementing that (simplest idea i have is using a tmp file to signal that startup is complete).

Some possible improvement I can think of
- Wait for a specified duration in `komorebic` before launching the bars
- Within the bar set a upper bound on the number of retries

But for both approaches finding the right duration/number of retries that caters to each user's machine might be tough.

What would be your approach to this problem?